jax/__init__.py

```diff
@@ -36,8 +36,11 @@
 # Confusingly there are two things named "config": the module and the class.
 # We want the exported object to be the class, so we first import the module
 # to make sure a later import doesn't overwrite the class.
-from jax import config as _config_module
-del _config_module
+import warnings as _warnings
+with _warnings.catch_warnings():
+  _warnings.simplefilter("ignore")
+  from jax import config as _config_module
+del _warnings, _config_module
```

Comment:
Eventually we'll need to have the config module in a non-public import path. Maybe cleaner to do that now: move `jax/config.py` to `jax/_src/config.py`, and then in `jax/config.py` use `_deprecation_getattr` to expose the old APIs while raising deprecation warnings. I think that's cleaner than ignoring the warning on import – what do you think?

Reply:
Oh wait, I see this import is doing something different than I thought. Hmm... maybe this is the best way for now.

Reply:
Question: if we always do this import in `jax/__init__.py`, is there any way for the user to actually see the warning?

Reply:
Huh, that is a great point. I guess the only other alternative is to set up a module `__getattr__` in `jax/config.py`?

Reply:
PTAL at the new version.